site/src/load.rs

```diff
@@ -213,6 +215,45 @@ impl SiteCtxt {
         )
     }
 
+    /// Returns the not yet tested published artifacts, sorted from oldest to newest.
+    pub async fn missing_artifacts(&self) -> anyhow::Result<Vec<String>> {
+        let artifact_list: String = reqwest::get("https://static.rust-lang.org/manifests.txt")
+            .await?
+            .text()
+            .await?;
+
+        lazy_static! {
+            static ref VERSION_REGEX: Regex = Regex::new(r"(\d+\.\d+.\d+)").unwrap();
+        }
+
+        let conn = self.conn().await;
+
+        let index = self.index.load();
+        let tested_artifacts: HashSet<_> = index.artifacts().collect();
+        let in_progress_artifacts: HashSet<_> = conn
+            .in_progress_artifacts()
+            .await
+            .into_iter()
+            .filter_map(|artifact| match artifact {
+                ArtifactId::Commit(_) => None,
+                ArtifactId::Tag(tag) => Some(tag),
+            })
+            .collect();
+
+        // Gather published artifacts that are not yet tested and are not in progress
+        let mut artifacts: Vec<String> = vec![];
+        for line in artifact_list.lines() {
+            if let Some(artifact) = parse_published_artifact_tag(&line) {
+                if !tested_artifacts.contains(artifact.as_str())
+                    && !in_progress_artifacts.contains(&artifact)
+                {
+                    artifacts.push(artifact);
+                }
+            }
+        }
+        Ok(artifacts)
```

Comment:
This should probably take only the last N artifacts we plan to consider, to avoid trying to benchmark 1.0 or similar (which won't work).

Reply:
I tried it locally and it starts with `beta-2016-03-23` :) And benchmarking that indeed doesn't work. I'll limit it to last 10 entries.